### PR TITLE
Remove isNative

### PR DESCRIPTION
**Pull request by conorcussell**
@ianstormtaylor I've [removed isNative](https://github.com/ianstormtaylor/slate/issues/963)

I guess the library benchmarks don't give us much info on the performance changes here, any tips on checking the performance of this in the browser?

```
benchmarks
    models
      get-blocks
        295.26 → 283.60 iterations/sec
      get-blocks-at-range
        2.73 → 2.80 iterations/sec
      get-characters
        86.72 → 100.63 iterations/sec
      get-characters-at-range
        7.47 → 8.48 iterations/sec
      get-inlines
        294.12 → 294.65 iterations/sec
      get-inlines-at-range
        2.64 → 2.68 iterations/sec
      get-marks
        17.04 → 17.32 iterations/sec
      get-marks-at-range
        5.04 → 4.42 iterations/sec
      get-path
        37.46 → 38.18 iterations/sec
      get-ranges
        1098.85 → 1056.25 iterations/sec
      get-texts
        259.95 → 244.42 iterations/sec
      get-texts-at-range
        95.68 → 97.23 iterations/sec
      update-descendant
        72.57 → 77.09 iterations/sec
    rendering
      normal
        1.23 → 1.25 iterations/sec
    serializers
      raw-deserialize
        5.64 → 5.63 iterations/sec
      raw-serialize
        62.00 → 59.60 iterations/sec
    transforms
      delete-backward
        246.63 → 285.42 iterations/sec
      insert-text
        285.92 → 289.88 iterations/sec
      normalize
        87.54 → 105.78 iterations/sec (21% faster)
      split-block
        69.83 → 83.92 iterations/sec (20% faster)
```

Also this appears to change the nature of at least one bug https://github.com/ianstormtaylor/slate/issues/938, instead of deleting the previous line, it does not allow you to add spaces on a wrapped line. I was working on a solution to this separately but still had some things to iron out.

### CODE REVIEW

**Comment by kgdev** [comment]
Why we can remove isNative? Does draftjs do so too?

File: src/plugins/core.js [hunk 1]
```diff
@@ -137,44 +134,9 @@ function Plugin(options = {}) {
     const nextText = next.startText
     const nextChars = nextText.getDecorations(decorators)
```
Comment (ianstormtaylor) [inline src/plugins/core.js]: I think `nextChars`, `nextText` (and probably others) are now unused so we don't need to compute them too? There might be a handful more lines of code we can remove here which will improve performance when checking further.

**Comment by ianstormtaylor** [comment]
@kgdev I'm not sure we can remove it, but it's a hunch I wanted to investigate. One thing we do which Draft doesn't is that we memoize the Immutable.js objects, so many of their methods are very low cost to call for the second time if nothing has changed. I have no idea if it's something we can avoid doing, but it was worth an investigation since `isNative` adds lots of complexity.

**Comment by conorcussell** [comment]
Thanks @ianstormtaylor I removed the rest of the code we don't need from that method. Will try and take a look at the perf impact in devtools and reply with my findings when I get a chance.

**Comment by ianstormtaylor** [comment]
Hey @conorcussell thanks again for kicking this off! I took your changes and added some more on #1088, so I'm going to close this now. Cheers!

**Pull request closed without merging**